addons/sale_project/models/project.py

```diff
@@ -173,6 +173,17 @@ def action_open_project_invoices(self):
             action['res_id'] = invoice_ids[0]
         return action
 
+    #todo retravailler l'action[views] atm makes no sense.
+    def action_aal(self, section_name, domain=None, res_id=False):
+        self.ensure_one()
+        if section_name in ['other_costs', 'other_revenues']:
+            action = self.env["ir.actions.actions"]._for_xml_id("analytic.account_analytic_line_action_entries")
+            action['domain'] = domain if domain else []
+            if res_id:
+                action['views'] = [(view_id, view_type) for view_id, view_type in action['views'] if view_type == 'form'] or [False, 'form']
```

Comment:
```suggestion
                action['views'] = [(view_id, view_type) for view_id, view_type in action['views'] if view_type == 'form'] or [(False, 'form')]
```
`action['views']` is a list of tuples right?